### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by Niki4tap**
Fancy logo, badges, who doesn't like it?

I'm not a web dev sorry, so if that html looks ugly, it probably is :sweat_smile: 

@xFrednet what do you think?

note: if you want to see how it looks, you can go here: https://github.com/Niki4tap/marker/tree/readme

(quick PR to forget about difficulties of improving `cargo-marker`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
+>
+> The project is in the very early stages of development, not everything is implemented yet.
+
+## Status
 
-The internal structure of marker and its components is documented in [`docs/internal/`](./docs/internal/).
+What works:
+- A few of example lints, which can be found in the `marker_lints` directory
+- Partial AST representation and API, both of which are in rapid development
+- Prototype `rustc` driver, to run the lints (being developed along with the API)
```
Comment (xFrednet) [inline README.md:23]: The wording here is a bit weird, I'd say either make it a bullet point list or whole sentences. What do you think about this:

```suggestion
- A basic Item and Pattern representation is available. Expressions and types are partially implement.
- Dynamic libraries can be loaded as *lint crates*. A few example lints, can be found in the `marker_lints` directory
- The API separated from the driver and translation layer. Rustc is currently the only backend development along with the API
```

File: README.md [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
+
+What doesn't work:
+- API is very much incomplete, but completing it is the main focus of the project right now
+- No integration with `rust-analyzer` or other drivers, only `rustc` as of now
+- ...and a lot more things! Check out [issues](https://github.com/rust-marker/marker/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) if you want to help
```
Comment (xFrednet) [inline README.md:28]: What do you think about this? I think it sounds more optimistic, but at the same time might be too much. I'm currently really hyped for the project and in life :3

```suggestion
What's next:
- The main goal is to complete the AST representation, and start on user tests
- A focus is also set on tooling. Working with marker should be simple and fun for lint developers and users.
- The project is designed to support multiple drivers and unify code analysis for Rust. Additional, drivers like `rust-analyzer` will be investigated.
- ...and a lot more things! Check out the open issues [issues](https://github.com/rust-marker/marker/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)

Sounds interesting?
* Do you have a lint idea? Help us out by creating a [user story](https://github.com/rust-marker/design/issues/new?assignees=&labels=A-user-story&template=user-story.md&title=)
* Have you found a bug or have suggestions? Please [create an issue](https://github.com/rust-marker/marker/issues/new), any kind of feedback or question is welcome!
* Do you want to help? Even better, check out [open issues](https://github.com/rust-marker/marker/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) or request help.
```

File: README.md [hunk 3]
```diff
@@ -1,17 +1,42 @@
...
 
-## Contributing
+[![Bors enabled](https://bors.tech/images/badge_small.svg)](https://app.bors.tech/repositories/46214)
+![CI Status](https://github.com/rust-marker/marker/actions/workflows/rust_bors.yml/badge.svg)
+
+**A project trying to implement a stable linting API for Rust.**
+
+> **Note**
+>
+> The project is in the very early stages of development, not everything is implemented yet.
```
Comment (xFrednet) [inline README.md:16]:
```suggestion
> The project is in the very early stages of development, some things are still missing and the API is still unstable.
```

**Review by xFrednet — COMMENTED** [review]
I'm a big fan! Updating the readme has been on my todo list, but you know how life is. This is a big help! We also got linked on the r/Rust subreddit. I was surprised by the number of stars and visitors the project got all of a sudden.

For the image. Would you mind moving it into a folder called `res` or `resources`? `msic` is so generic that I fear that things will collect there over time. I've also attached the SVG file, for better resolution and to safe it somewhere besides my computer ^^

**Comment by xFrednet** [comment]
https://user-images.githubusercontent.com/17087237/215619629-afb5c6a5-a5bc-46ee-894d-2766392bf52a.svg

File: README.md [hunk 4]
```diff
@@ -1,17 +1,42 @@
...
+</p>
 
-A project trying to implement a stable linting API for Rust.
+<br/>
 
-## Contributing
+[![Bors enabled](https://bors.tech/images/badge_small.svg)](https://app.bors.tech/repositories/46214)
+![CI Status](https://github.com/rust-marker/marker/actions/workflows/rust_bors.yml/badge.svg)
+
+**A project trying to implement a stable linting API for Rust.**
```
Comment (xFrednet) [inline README.md:12]: I just saw that this is different than the project description on GitHub.

> An experimental linting interface for Rust. Let's make custom lints a reality!

Which one do you like better?

**Comment by Niki4tap** [comment]
I'm really not sure about that last comment, I think both are fine, but I've applied and pushed all the suggestions, just to see how that would look like, I think I like it, see if you want to change anything else, nothing stands out to me at the moment

File: README.md [hunk 5]
```diff
@@ -1,17 +1,48 @@
...
+> **Note**
+>
+> The project is in the very early stages of development, some things are still missing and the API is not stable yet.
+
+## Status
+
+What works:
+- A basic Item and Pattern representation is available. Expressions and types are partially implemented
+- Dynamic libraries can be loaded as *lint crates*. A few example lints can be found in the `marker_lints` directory
+- The API separated from the driver and translation layer. Rustc is currently the only backend development along with the API
```
Comment (xFrednet) [inline README.md:23]:
```suggestion
- The API is separated from the driver and translation layer. Rustc is currently the only backend development along with the API
```

File: README.md [hunk 6]
```diff
@@ -1,17 +1,48 @@
...
 
-The internal structure of marker and its components is documented in [`docs/internal/`](./docs/internal/).
+What's next:
+- The main goal is to complete the AST representation, and start on user tests
+- A focus is also set on tooling. Working with marker should be simple and fun for lint developers and users.
+- The project is designed to support multiple drivers and unify code analysis for Rust. Additionally, drivers like `rust-analyzer` will be investigated.
+- ...and a lot more things! Check out the open issues [issues](https://github.com/rust-marker/marker/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)
```
Comment (xFrednet) [inline README.md:29]:
```suggestion
- ...and a lot more things! Check out the open [issues](https://github.com/rust-marker/marker/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc)
```

**Review by xFrednet — APPROVED** [review]
Thank you, I like this version! Two small nits, (coming from my suggestion 😅). 

Then it'll be perfect! You can `r=me` afterwards.

**Comment by Niki4tap** [comment]
bors r=xFrednet

**Pull request merged**